src/controller/CHIPDevice.cpp

```diff
@@ -470,5 +496,29 @@ void Device::AddReportHandler(EndpointId endpoint, ClusterId cluster, AttributeI
     mCallbacksMgr.AddReportCallback(mDeviceId, endpoint, cluster, attribute, onReportCallback);
 }
 
+Device::~Device()
```

Comment:
I would request that this device self-persistence be refactored as a method on the device object that the device simply calls in its destructor.

Right now, we have device persistence code duplicated here and in CHIPDeviceController.cpp.